### PR TITLE
Cluster spec docs for EKS-A cluster on CloudStack

### PR DESCRIPTION
**Pull request by chrisnegus**
*Issue #, if available:* [#3658](https://github.com/aws/eks-anywhere/issues/3658)

*Description of changes:*
Documentation for clusterspec file to create EKS-A cluster on Cloudstack.

### CODE REVIEW

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:238]: From the description, should this be rewritten as
```suggestion
### availabilityZones.zone.name (required)
```

Note that a user can also pass in `availabilityZones.zone.id` instead.

There should also be a separate line item for `availabilityZones.zone` which can either contain a name or an id

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:266]:
```suggestion
The VM template to use for your EKS Anywhere cluster. Currently, a VM based on RHEL 8.6 is required.
```

**Review by maxdrib — CHANGES_REQUESTED** [review]

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:93]: I added those fields.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:106]: Done.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:259]: There's a special contract here between EKS-A and CAPC. If the user provides the label here `ds.meta_data.failuredomain` then CAPC will substitute the value to provide the actual AZ name as a label on the worker nodes. @wanyufe has the lower level details if you have questions

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:267]: Okay. I made it optional and pointed to the description of adding the credentials in the getting started guide. Let me know if anything further is needed.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:270]: Done.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:281]: I think I get what you are saying. Please check once I commit this.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:287]: Done

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:288]: Done.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:308]: Done.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:309]: Done

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:265]: Okay. I talked to @wanyufe about this. I added information here and to the control plane labels. Please check that I got the wording right.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md]: Done.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:284]: This is required. You can specify either the profile name (from ini file) or the previously created secret on the cluster, in the `eksa-system` namespace

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:289]: this is optional, it should default to ROOT

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:297]:
```suggestion
### availabilityZones.name (required)
Unique name to categorize the availability zone in which the cluster will reside
```

zone name/id are specified at availabilityZones.zone.{id,name}

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:267]: I'm not sure if "for example" makes sense here, since this keyword `ds.meta_data.failuredomain` is specifically reserved for the availability zones replacement usecase. It's more of a custom feature than a generic example

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:281]: I'm actually not sure if the default is admin. I think the default would be a lookup to whatever user the api key and secret key were generated from. I'd remove the point about defaulting.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:299]: I don't think network *name* is required specifically. You have to specify a network. To do that, you can _either_ provide `name` or `id`. Similar to `computeOffering` below

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:305]:
```suggestion
### computeOffering (required)
```

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:308]: I think you can pass in an arbitrary number of user objects, not just 1

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:328]: to be clear, it would look like
```
template:
  name: xxxx
```

or 

```
template:
  id: yyyy
```

so one of template.name or template.id are required. Same for a lot of the other fields that reference CloudStack resources

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:331]: diskOffering.name?

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:334]: diskOffering.mountPath? All the diskoffering attributes should be nested under diskOffering

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (wanyufe) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:71]: A special label value is supported by CAPC provider: 

```
labels:
      cluster.x-k8s.io/failure-domain: ds.meta_data.failuredomain
```
`ds.meta_data.failuredomain` will be replaced with a failuredomain name where the node is deployed in, for example `az-1`

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:267]: Okay. I think it's the only custom label as well. I'll remove "For example".

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:349]: I think we're also missing the optional field `affinity` which enables a user to set the `pro` or `anti`, and create and manage an affinity group on their behalf. The `affinity` field can be used in a mutually exclusive fashion with the `affinityGroupIDs` field

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (wanyufe) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:48]: A special label value is supported by CAPC provider:

```
labels:
      cluster.x-k8s.io/failure-domain: ds.meta_data.failuredomain
```

`ds.meta_data.failuredomain` will be replaced with a failuredomain name where the node is deployed in, for example `az-1`

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (maxdrib) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:345]: could you add the optional `userCustomDetails` field as well?

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:48]: I like that wording. Done.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:71]: Done.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:267]: I reworded base on comments above.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:281]: Removed.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:284]: Done.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:289]: Got it. Thanks.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:299]: Done.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (wanyufe) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:281]: If account is not specified and there are more than one account under the domain, CAPC will throw an error. If there is only one account, it will use it -- it may not be `admin`

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (wanyufe) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:267]: Customer can add other labels like `foo: bar` and nodes will be labeled like that.

`ds.meta_data.failuredomain` is a special keyword case Cloudstack supported.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:308]: Done.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:328]: I used the {id,name} notation on each field it applies to. I'll go through later and see if I missed any.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:334]: Done.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:331]: Done.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:345]: I added it. It looks like a way of adding key/value pairs to nodes in a CloudStackMachineConfig. Is there some purpose I could give as an example?

Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:345]: Got it. Thanks.

File: docs/content/en/docs/reference/clusterspec/cloudstack.md
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/cloudstack.md:354]: Not sure I got this right. Let's revisit after everything else is done.

**Review by vignesh-goutham — APPROVED** [review]
/approve

**Comment by chrisnegus** [comment]
I fixed a broken link, so it needs another lgtm if it looks okay to merge.

**Review by vignesh-goutham — APPROVED** [review]
/approve

**Pull request merged**
